src/components/cesium_map/cesium_UI.js

```diff
@@ -389,15 +382,12 @@ class CesiumMap extends React.Component {
   // https://medium.com/@garrettmac/reactjs-how-to-safely-manipulate-the-dom-when-reactjs-cant-the-right-way-8a20928e8a6
   // manipulate Dom outside the react model
   shouldComponentUpdate(nextProps) {
-    // update point primitive based on searchFields
```

Comment:
In the previous old code, whenever the parent component (iSamples_resultList) state is changed (e.g. the pending flag that changes whenever we do a new query using the facet fields), this caused the child component CesiumMap to be re-rendered. As result, this shouldComponentUpdate was not invoked at all - only the componentDidMount was called as the child component was constantly mounted and unmounted whenever the state of the parent component was called.